api/handlers_e2e_test.go

```diff
@@ -300,6 +302,98 @@ func TestPagingRootTxnDeduplication(t *testing.T) {
 	}
 }
 
+func TestTransactionHandler(t *testing.T) {
+	db, shutdownFunc := setupIdb(t, test.MakeGenesis(), test.MakeGenesisBlock())
+	defer shutdownFunc()
+
+	///////////
+	// Given // A block containing a key reg txn with state proof key
+	///////////
+	var votePK crypto.OneTimeSignatureVerifier
+	votePK[0] = 1
+
+	var selectionPK crypto.VRFVerifier
+	selectionPK[0] = 1
+
+	var sprfkey merklekeystore.Verifier
+	sprfkey[0] = 1
+
+	txn := transactions.SignedTxnWithAD{
+		SignedTxn: transactions.SignedTxn{
+			Txn: transactions.Transaction{
+				Type: "keyreg",
+				Header: transactions.Header{
+					Sender:      test.AccountA,
+					GenesisHash: test.GenesisHash,
+				},
+				KeyregTxnFields: transactions.KeyregTxnFields{
+					VotePK: votePK,
+					SelectionPK:selectionPK,
+					StateProofPK: sprfkey,
```

Comment:
It would be nice to rename `sprfkey` to `stateProofPK` for consistency with `votePK` and `selectionPK`.

Reply:
updated.